components/brave_vpn/browser/vpn_service.h

```diff
@@ -0,0 +1,112 @@
+
+  void GetAllServerRegions(ResponseCallback callback);
+  void GetTimezonesForRegions(ResponseCallback callback);
+  void GetHostnamesForRegion(ResponseCallback callback,
+                             const std::string& region);
+  void GetSubscriberCredential(ResponseCallback callback,
+                               const std::string& product_type,
+                               const std::string& product_id,
+                               const std::string& validation_method,
+                               const std::string& purchase_token);
+  void VerifyPurchaseToken(ResponseCallback callback,
+                           const std::string& purchase_token,
+                           const std::string& product_id,
+                           const std::string& product_type);
+
+ private:
+  static GURL oauth_endpoint_;
+  static GURL api_endpoint_;
+  using SimpleURLLoaderList =
+      std::list<std::unique_ptr<network::SimpleURLLoader>>;
+
+  using URLRequestCallback =
+      base::OnceCallback<void(const int,
+                              const std::string&,
+                              const std::map<std::string, std::string>&)>;
+
+  bool OAuthRequest(const GURL& url,
+                    const std::string& method,
+                    const std::string& post_data,
+                    bool set_app_ident,
+                    URLRequestCallback callback);
+  void OnURLLoaderComplete(SimpleURLLoaderList::iterator iter,
+                           URLRequestCallback callback,
+                           const std::unique_ptr<std::string> response_body);
+
+  void OnGetAllServerRegions(ResponseCallback callback,
+                             const int status,
+                             const std::string& body,
+                             const std::map<std::string, std::string>& headers);
+
+  void OnGetTimezonesForRegions(
+      ResponseCallback callback,
+      const int status,
+      const std::string& body,
+      const std::map<std::string, std::string>& headers);
+
+  void OnGetHostnamesForRegion(
+      ResponseCallback callback,
+      const int status,
+      const std::string& body,
+      const std::map<std::string, std::string>& headers);
+
+  void OnGetSubscriberCredential(
+      ResponseCallback callback,
+      const int status,
+      const std::string& body,
+      const std::map<std::string, std::string>& headers);
+
+  void OnVerifyPurchaseToken(ResponseCallback callback,
+                             const int status,
+                             const std::string& body,
+                             const std::map<std::string, std::string>& headers);
+
+  content::BrowserContext* context_;
+  scoped_refptr<network::SharedURLLoaderFactory> url_loader_factory_;
+  SimpleURLLoaderList url_loaders_;
+  base::WeakPtrFactory<VpnService> weak_factory_;
+  base::WeakPtrFactory<VpnService> vpn_weak_factory_{this};
+
+  DISALLOW_COPY_AND_ASSIGN(VpnService);
```

Comment:
This define is deprecated - https://source.chromium.org/chromium/chromium/src/+/main:base/macros.h;l=13